lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -372,32 +372,57 @@ def add_test_targets
             end
           end
 
+          # Attaches an app host target to a given test target
+          #
+          # @param [PBXNativeTarget] app_host_target
+          # @param [PBXNativeTarget] test_native_target
+          #
+          def attach_app_host_target_to_test_native_target(app_host_target, test_native_target)
+            test_native_target.build_configurations.each do |configuration|
+              test_host = "$(BUILT_PRODUCTS_DIR)/#{app_host_target.name}.app/"
+              test_host << 'Contents/MacOS/' if target.platform == :osx
+              test_host << app_host_target.name.to_s
+              configuration.build_settings['BUNDLE_LOADER'] = '$(TEST_HOST)'
+              configuration.build_settings['TEST_HOST'] = test_host
+            end
+            target_attributes = project.root_object.attributes['TargetAttributes'] || {}
+            target_attributes[test_native_target.uuid.to_s] = { 'TestTargetID' => app_host_target.uuid.to_s }
+            project.root_object.attributes['TargetAttributes'] = target_attributes
+            test_native_target.add_dependency(app_host_target)
+          end
+
           # Adds the test app host targets for the library to the Pods project with the
           # appropriate build configurations.
           #
           # @param  [Array<PBXNativeTarget>] test_native_targets
           #         the test native targets that have been created to use as a lookup when linking the app host to.
+          # @param  [Array<PBXNativeTarget>] app_native_targets
+          #         the app native targets that have been created to use as a lookup when linking the app host to.
           #
           # @return [Array<PBXNativeTarget>] the app host targets created.
           #
-          def add_test_app_host_targets(test_native_targets)
-            target.test_spec_consumers.select(&:requires_app_host?).group_by(&:test_type).map do |test_type, test_spec_consumers|
+          def add_test_app_host_targets(test_native_targets, app_native_targets)
+            target.test_spec_consumers.reject(&:requires_app_host?).select(&:app_host_name).each do |test_spec_consumer|
+              raise "`#{target.label}-#{test_spec_consumer.test_type}-Tests` manually specifies an app host but has `test_spec.requires_app_host = false`! Please set `test_spec.requires_app_host = true`"
+            end
+
+            target.test_spec_consumers.select(&:requires_app_host?).select(&:app_host_name).each do |test_spec_consumer|
+              name = test_spec_consumer.app_host_name.split('/').join('-')
+              app_native_target = app_native_targets.find { |target| target.name == name }
```

Comment:
will `app_native_targets` here have app targets from other PodTarget instances or no?

Reply:
I believe it should. I'm working on figuring that out all the way

Reply:
It looks like it does not